framework/src/modules/chain/transactions/votes_weight.js

```diff
@@ -206,30 +206,35 @@ const undo = (stateStore, transaction, exceptions = {}) => {
 
 const prepare = async (stateStore, transaction) => {
 	// Get delegate public keys whom sender voted for
-	const senderDelegatePks =
+	const senderDelegatePublicKeys =
 		stateStore.account.getOrDefault(transaction.senderId)
 			.votedDelegatesPublicKeys || [];
 
 	const recipientId = getRecipientAddress(stateStore, transaction);
 
 	// Get delegate public keys whom recipient voted for
-	const recipientDelegatePks =
+	const recipientDelegatePublicKeys =
 		(recipientId &&
 			stateStore.account.getOrDefault(recipientId).votedDelegatesPublicKeys) ||
 		[];
 
 	// Get unique public keys from merged list
-	const uniqPksToBeCached = [
-		...new Set([...senderDelegatePks, ...recipientDelegatePks]),
+	const uniqPublicKeysToBeCached = [
+		...new Set([...senderDelegatePublicKeys, ...recipientDelegatePublicKeys]),
 	];
 
-	const addressesToBeCached = uniqPksToBeCached
-		// format items for filtering in cache function
-		.map(delegatePk => ({
-			address: getAddressFromPublicKey(delegatePk),
-		}));
-
-	return stateStore.account.cache(addressesToBeCached);
+	/**
+	 * We are running `stateStore.account.cache` multiple times per public key instead of once with array of public keys
+	 * because StateStore uses `storage.entities.Account.get` which default limit result is 101 entries
+	 * meaning it will only cache maximum 101 accounts.
+	 */
+	return Promise.all(
+		uniqPublicKeysToBeCached
+			.map(delegatePublicKey => ({
+				address: getAddressFromPublicKey(delegatePublicKey),
+			}))
+			.map(filter => stateStore.account.cache(filter)),
+	);
```

Comment:
This change comes with a big performance degradation. This will result into minimum 1 to 34 public keys. So in worst case scenario you are converting 1 SQL query to 34 different queries. That I would not recommend to do. 

Apart from this change, this line actually not makes any sense to SDK developers.
https://github.com/LiskHQ/lisk-sdk/blob/3a866573edf27f824837cc478c0abdd2aacee264/framework/src/modules/chain/state_store/account_store.js#L53

For a custom transaction developer, they can use `cache(filters)` in the `prepare` step. And what ever matching records will come for `filters` they will be cached. We had not documented anywhere that it will not cache more than 101 accounts, and which actually does not makes sense in the prepare steps. 

So two solutions can be done. 
1. Add `limit: null` inside cache method and add condition that user must provide `filters` for the cache. 
2. Or add `options` as second parameter to `cache` method so user can set himself what options needed for storage.

Reply:
After discussion with the team the `limit: null` solution was chosen.

Reply:
After discussion with the team the limit: null solution was chosen.